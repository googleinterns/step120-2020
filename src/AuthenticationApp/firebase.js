import firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import "firebase/analytics";
import APIKey from "./APIKey";
 
const firebaseConfig = {
    apiKey: APIKey,
    authDomain: "step120-2020.firebaseapp.com",
    databaseURL: "https://step120-2020.firebaseio.com",
    projectId: "step120-2020",
    storageBucket: "step120-2020.appspot.com",
    messagingSenderId: "994372108603",
    appId: "1:994372108603:web:cb5d95fa2ec4d107b1b71e",
    measurementId: "G-C0ZHMWXRWH"
  };
 
  firebase.initializeApp(firebaseConfig);
  //firebase.analytics();
  export const generateUserDocument = async (user, furtherData) => {
      if(!user) return;
      
      //fill doc with user data
      
      return getUserDocument(user.uid);
  };
  const getUserDocument = async uid => {
      if (!uid) return null;
      try {
          const userDocument = await firestore.doc(`users/${uid}`).get();
          return {
              uid,
              ...userDocument.data()
          };
      } catch (error) {
          console.error("Error fetching user", error);
      }
  };
  export const authentication = firebase.auth();
  export const firestore = firebase.firestore();
