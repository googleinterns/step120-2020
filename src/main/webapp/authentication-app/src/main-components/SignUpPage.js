import React, { useState } from "react";
import { Link } from "@reach/router";

const SignUpPage = () => {
    const [userEmail, setUserEmail] = useState("");
    const [userPassword, setUserPassword] = useState("");
    const [displayName, setDisplayName] = useState("");
    const [error, setError] = useState(null);

    const createUser = (event, userEmail, userPassword) => {
        event.preventDefault();
        setUserEmail("");
        setUserPassword("");
        setDisplayName("");
    };

    const onChangehandler = event => {
        const { name, value } = event.currentTarget;
        if( name === userPassword) {
            setUserPassword(value);
        }
        else if (name === displayName) {
            setDisplayName(value);
        }
        else if (name === userEmail) {
            setUserEmail(value);
        }
    };

    return (
        <div className="SignUp">
        <h1 className="Title">Sign Up </h1>
        <div className="SignUp-Area">
          <form className="SignUp-form">
            <label htmlFor="displayName" className="block">
              Choose your Display Name:
            </label>
            <input
              type="text"
              className=""
              name="displayName"
              value={displayName}
              placeholder= "E.x: AwesomeDude123"
              id="displayName"
              onChange={event => onChangeHandler(event)}
            />
            <label htmlFor="userEmail" className="block">
              Your Email:
            </label>
            <input
              type="userEmail"
              className=""
              name="userEmail"
              value= {userEmail}
              placeholder= "E.x: JohnAppleseed@gmail.com"
              id="userEmail"
              onChange = {(event) => onChangeHandler(event)} 
            />
            <label htmlFor="userPassword" className="block">
              Make a Strong Password:
            </label>
            <input
              type="userPassword"
              className=""
              name="userPassword"
              value= {userPassword}
              placeholder="Your Password Here"
              onChange = {(event) => onChangeHandler(event)}
            />
            <button 
              className="signUpbutton"
              onClick= { event => {createUser(event, userEmail, userPassword)}}>
              Click to Sign Up
            </button>
          </form>
          <p className="">or</p>
          <button> Sign in with Google </button>
          <p className="">
            Already have an Account? Sign In! {" "}
            <Link to = "/" className="signin-link">
              Sign up 
            </Link>{" "}
          </p>
        </div>
      </div>
    )
}
export default SignUpPage;

