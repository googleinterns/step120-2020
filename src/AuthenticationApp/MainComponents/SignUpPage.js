import React, { useState } from "react";
import { Link, navigate } from "@reach/router";
import { authentication, generateUserDocument } from "../firebase";
 
const SignUpPage = () => {
    const [userEmail, setUserEmail] = useState("");
    const [userPassword, setUserPassword] = useState("");
    const [retypedPassword, setRetypedPassword] = useState("");
    const [displayName, setDisplayName] = useState("");
    const [error, setError] = useState(null);
 
    const createUser = async (
        event, 
        userEmail, 
        userPassword, 
        displayName,
        retypedPassword
        ) => {
            event.preventDefault();
            
            try {
                if (retypedPassword !== userPassword) {
                    throw new SyntaxError("Your passwords don't match");
                } else {  
                    const {user} = await authentication.createUserWithEmailAndPassword(
                        userEmail, 
                        userPassword
                        );
                    user.updateProfile({
                        displayName: displayName
                    }).then(() => {
                        console.log("update successful");
                    })
                    generateUserDocument(user);
                }
            } catch(error) {
                setError('Error signing up with email and password, ' + error.message);
            }
 
            setUserEmail("");
            setUserPassword("");
            setDisplayName("");
            setRetypedPassword("");
            if (authentication.currentUser) {
                navigate("/");
            }
        };
 
    const onUserPasswordChangeHandler = event => {
        const {value} = event.currentTarget;
        setUserPassword(value);
    }
    const onDisplayNameChangeHandler = event => {
        const {value} = event.currentTarget;
        setDisplayName(value);
    }
    const onUserEmailChangeHandler = event => {
        const {value} = event.currentTarget;
        setUserEmail(value);
    }
    const onRetypedPasswordChangeHandler = event => {
        const {value} = event.currentTarget;
        setRetypedPassword(value);
    }
 
    return (
        <div className="SignUp">
        <h1 className="Title">Sign Up </h1>
        <div className="SignUp-Area">
          {error !== null && <div>{error}</div>}
          <form className="SignUp-form">
            <label htmlFor="displayName" className="block">
              Choose your Display Name:
            </label>
            <input
              type="text"
              className=""
              name="displayName"
              value={displayName}
              placeholder= "E.x: AwesomeDude123"
              id="displayName"
              onChange={(event) => onDisplayNameChangeHandler(event)}
            />
            <label htmlFor="userEmail" className="block">
              Your Email:
            </label>
            <input
              type="userEmail"
              className=""
              name="userEmail"
              value= {userEmail}
              placeholder= "E.x: JohnAppleseed@gmail.com"
              id="userEmail"
              onChange = {(event) => onUserEmailChangeHandler(event)} 
            />
            <label htmlFor="userPassword" className="block">
              Make a Strong Password:
            </label>
            <input
              type="password"
              className=""
              name="userPassword"
              value= {userPassword}
              placeholder="Your Password Here"
              onChange = {(event) => onUserPasswordChangeHandler(event)}
            />
            <label htmlFor="retypedPassword" className="block">
              Retype your Password:
            </label>
            <input
              type="password"
              className=""
              name="retypedPassword"
              value= {retypedPassword}
              placeholder="Type your password again"
              onChange = {(event) => onRetypedPasswordChangeHandler(event)}
            />
            <button 
              className="signUpbutton"
              onClick= { event => {createUser(
                  event, 
                  userEmail, 
                  userPassword, 
                  displayName,
                  retypedPassword
                  )}}>
              Click to Sign Up
            </button>
          </form>
          <p className="">or</p>
          <button> Sign in with Google </button>
          <p className="">
            Already have an Account? Sign In! {" "}
            <Link to = "/signIn" className="signin-link">
              Sign In 
            </Link>
          </p>
        </div>
      </div>
    );
}
export default SignUpPage;
