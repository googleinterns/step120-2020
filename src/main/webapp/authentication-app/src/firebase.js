import firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";

const firebaseConfig = {
    apiKey: "AIzaSyAjse2YjCJHUUe5w1i7I9jkjH8cqqRcAIU",
    authDomain: "step120-2020.firebaseapp.com",
    databaseURL: "https://step120-2020.firebaseio.com",
    projectId: "step120-2020",
    storageBucket: "step120-2020.appspot.com",
    messagingSenderId: "994372108603",
    appId: "1:994372108603:web:cb5d95fa2ec4d107b1b71e",
    measurementId: "G-C0ZHMWXRWH"
  };

  firebase.initializeApp(firebaseConfig);
  firebase.analytics();
  export const authentication = firebase.auth();
  export const firestore = firebase.firestore();