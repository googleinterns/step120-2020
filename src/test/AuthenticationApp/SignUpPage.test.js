import * as firebase from 'firebase';
import React from 'react';
import ReactDOM from 'react-dom';
import SignUpPage from '../../AuthenticationApp/MainComponents/SignUpPage';
import { shallow } from "enzyme";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import renderer from 'react-test-renderer'

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });


describe('<SignUpPage />', () => {
    let wrapper;
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState]);

    afterEach(() => {
        jest.clearAllMocks();
    });

    describe("Snapshot Test" , () => {
        it("should render without crashing", () => {
            const domDiv = document.createElement('div');
            ReactDOM.render(<SignUpPage />, domDiv);
        });

        it("renders a snapshot of Component", () => {
            const snapshot = renderer.create(<SignUpPage />).toJSON();
            expect(snapshot).toMatchSnapshot();
        })

    })

    it("should update userPassword state", () => {
        wrapper = mount(<SignUpPage onUserPasswordChangeHandler={setState} />);
        
        wrapper.find('.userPassword').simulate('change');
        expect(setState).toBeTruthy();
    });

    it("should update userEmail state", () => {
        wrapper = mount(<SignUpPage onUserEmailChangeHandler={setState} />);
        
        wrapper.find('.userEmail').simulate('change');
        expect(setState).toBeTruthy();
    });

    it("should update displayName state", () => {
        wrapper = mount(<SignUpPage onDisplayNameChangeHandler={setState} />);
        
        wrapper.find('.displayName').simulate('change');
        expect(setState).toBeTruthy();
    });

    it("should update retypedPassword state", () => {
        wrapper = mount(<SignUpPage onRetypedPasswordChangeHandler={setState} />);
        
        wrapper.find('.retypedPassword').simulate('change');
        expect(setState).toBeTruthy();
    });
    
});
    

