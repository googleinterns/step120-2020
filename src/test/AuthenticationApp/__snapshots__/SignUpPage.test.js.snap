// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage /> Snapshot Test renders a snapshot of Component 1`] = `
<div
  className="SignUp"
>
  <h1
    className="Title"
  >
    Sign Up 
  </h1>
  <div
    className="SignUp-Area"
  >
    <form
      className="SignUp-form"
    >
      <label
        className="block"
        htmlFor="displayName"
      >
        Choose your Display Name:
      </label>
      <input
        className="displayName"
        id="displayName"
        name="displayName"
        onChange={[Function]}
        placeholder="E.x: AwesomeDude123"
        type="text"
        value=""
      />
      <label
        className="block"
        htmlFor="userEmail"
      >
        Enter Your .edu Email:
      </label>
      <input
        className="userEmail"
        id="userEmail"
        name="userEmail"
        onChange={[Function]}
        placeholder="E.x: JohnAppleseed@gmail.edu"
        type="userEmail"
        value=""
      />
      <label
        className="block"
        htmlFor="userPassword"
      >
        Make a Strong Password:
      </label>
      <input
        className="userPassword"
        name="userPassword"
        onChange={[Function]}
        placeholder="Your Password Here"
        type="password"
        value=""
      />
      <label
        className="block"
        htmlFor="retypedPassword"
      >
        Retype your Password:
      </label>
      <input
        className="retypedPassword"
        name="retypedPassword"
        onChange={[Function]}
        placeholder="Type your password again"
        type="password"
        value=""
      />
      <button
        className="signUpButton"
        onClick={[Function]}
      >
        Click to Sign Up
      </button>
    </form>
    <p
      className=""
    >
      or
    </p>
    <button>
       Sign in with Google 
    </button>
    <p
      className=""
    >
      Already have an Account? Sign In! 
       
      <a
        className="signin-link"
        href="/signIn"
        onClick={[Function]}
      >
        Sign In
      </a>
    </p>
  </div>
</div>
`;
